### PR TITLE
IGVF-922-lib-typescript

### PR DESCRIPTION
**Pull request by dougli1sqrd**
* lib/general.ts
* lib/color.ts

to get started

### CODE REVIEW

File: lib/color.ts [hunk 1]
```diff
@@ -7,7 +13,7 @@
  * @return {object.g} Green component as an integer from 0 to 255
  * @return {object.b} Blue component as an integer from 0 to 255
```
Comment (forresttanaka) [inline lib/color.ts:14]: Change line 11 from `{object}` to `{Color}`.

File: lib/color.ts [hunk 2]
```diff
@@ -23,7 +29,7 @@ export function colorToTriple(color) {
  * @param {string} color CSS hex color to test
  * @return {bool} True if `color` is a "light" color.
```
Comment (forresttanaka) [inline lib/color.ts:30]: Change to `boolean`. I used to use `bool` because that’s what `PropTypes` uses.

File: lib/general.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
  * @param {string} path The @id of the object to get the type for.
  * @returns Type of the object this path comes from; or the empty string if no type could be found.
```
Comment (forresttanaka) [inline lib/general.ts:4]: Hmm. I forgot to have a type here. So `@returns {string}`.

File: lib/color.ts [hunk 4]
```diff
@@ -1,3 +1,9 @@
+interface Color {
```
Comment (forresttanaka) [inline lib/color.ts:1]: I think to call this function from Typescript, you might need to export this interface. I’m about 60% sure of that. You could try it by converting the color.test.js Jest test to Typescript.

File: lib/general.ts [hunk 5]
```diff
@@ -26,13 +26,23 @@ export function isValidUrl(url) {
...
   return url.split("?")[0];
 }
 
+type JSON =
+  | null
+  | boolean
+  | string
+  | number
+  | Array<JSON>
+  | { [key: string]: JSON };
```
Comment (forresttanaka) [inline lib/general.ts:44]: I wonder if this should go in globals.d.ts? Not sure if this type is useful outside of this function or not.

**Review by forresttanaka — CHANGES_REQUESTED** [review]
Most of this is about comments. I do have a suggestion about moving types/interfaces around, but I’m really not sure if that’s needed or not. But it’s easy to test by converting the corresponding Jest tests to Typescript.

File: lib/general.ts (hunk 5 above)
Comment (dougli1sqrd) [reply lib/general.ts:44]: Yeah not sure either. I was going to keep it this way for now, and as we converted more if I needed `JSON` I would move to a better location

File: lib/color.ts (hunk 4 above)
Comment (dougli1sqrd) [reply lib/color.ts:1]: This looks like it works okay. I'll commit the test as ts as well

**Comment by forresttanaka** [comment]
@dougli1sqrd Lint error in a couple files.

**Review by forresttanaka — APPROVED** [review]
😎

**Pull request merged**